src/server/snap.rs

```diff
@@ -191,10 +208,30 @@ pub fn send_snap(
     let (sink, receiver) = client.snapshot()?;
 
     let send_task = async move {
-        let mut sink = sink.sink_map_err(Error::from);
-        sink.send_all(&mut chunks).await?;
-        sink.close().await?;
-        let recv_result = receiver.map_err(Error::from).await;
+        let send_and_recv = async {
+            let mut sink = sink.sink_map_err(Error::from);
+
+            #[cfg(feature = "failpoints")]
+            {
+                let should_delay = (|| {
+                    fail::fail_point!("snap_send_timer_delay", |_| { true });
+                    false
+                })();
+                if should_delay {
+                    _ = GLOBAL_TIMER_HANDLE
+                        .delay(StdInstant::now() + Duration::from_secs(1))
+                        .compat()
+                        .await;
+                }
+            }
+            sink.send_all(&mut chunks).await?;
+            sink.close().await?;
+            Ok(receiver.map_err(Error::from).await)
+        };
+        let sink_timeout = TimeoutFuture::new(send_and_recv, get_snap_timeout(), || {
+            Err(Error::Other(box_err!("send snapshot timeout")))
```

Comment:
Could you add a metric or log here?

Reply:
There is error log in the caller of this function.